src/blocksender.cpp

```diff
@@ -84,21 +98,28 @@ void BlockSender::sendBlock(CNode& node,
 
     // We only support MSG_XTHINBLOCK, if peer wants MSG_THINBLOCK,
     // fallback to full one.
-    if (invType == MSG_BLOCK || invType == MSG_THINBLOCK)
+    if (invType == MSG_BLOCK || (invType == MSG_THINBLOCK
+                && !NodeStatePtr(node.id)->supportsCompactBlocks))
```

Comment:
I mean, sending a MSG_THINBLOCK INV - it could be requesting a THINBLOCK _and_ supporting Compact Blocks.